astropy/coordinates/distances.py

```diff
@@ -153,37 +152,33 @@ def __new__(
                 # the log of the distance
```

Comment:
While we are at it, how about changing the comment, because what is here suggests we are guessing the unit of the input, while really the value has a unit, but we're just converting it:
```
If no explicit unit was given, convert to a suitable one
based on the mean of the log of the distance. 
# Maybe also add?? (This since I wondered about what happened later)
We leave the unit at None so that no further conversion/copying
is done below.
```
This also works better with actually converting `value`, as you do now -- much nicer!